fms/utils/serialization.py

```diff
@@ -114,25 +149,32 @@ def get_ckp_format(model_path: Union[str, Path]) -> str:
 
 def load_state_dict(
     model_path: Union[str, Path],
+    model: torch.nn.Module,
+    architecture: str,
+    source: str,
     *,
     checkpoint_format: Optional[str] = None,
     distributed_strategy: Optional[str] = None,
     checkpoint_sharding: Optional[str] = None,
     initial_device: torch.device = torch.device("cpu"),
     rank: int = 0,
     world_size: int = 1,
-) -> Mapping[str, Any]:
+) -> torch.nn.Module:
     """
     Validates that the file(s) found at a checkpoint path are compatible with
-    the intended (possibly distributed) use-case, and returns the state dict
-    if needed.
+    the intended (possibly distributed) use-case, and returns the model with
+    the weights loaded on it.
```

Comment:
it does mutate the model, so even if the user ignored the return type the model would be updated. In the invalid case from validation though, this is a no-op and (would presumably) return the original model

Reply:
added a comment about memory efficiency to justify the in-place mutation